aodntools/timeseries_products/Documentation/velocity_hourly_timeseries.md

```diff
@@ -0,0 +1,125 @@
+- File is a delayed-mode, quality-controlled product (file version label “FV01”);
+- File is compliant with CF-1.6 and IMOS-1.4 conventions;
+- File contains, at the minimum, the components of current velocity (`UCUR`, `VCUR`), and variables `TIME`, `DEPTH`, `LATITUDE`, `LONGITUDE`, and `HEIGHT_ABOVE_SENSOR` in the case of ADCPs;
+- All files to be aggregated are from the same site, and have the same `site_code` attribute; 
+- Variables to be aggregated have `TIME` and (optionally) `HEIGHT_ABOVE_SENSOR` as their only dimensions (or if `LATITUDE` and `LONGITUDE` are included as dimensions, they have size 1);
+- The in-water data are bounded by the global attributes `time_deployment_start` and `time_deployment_end`;
+
+
+
+The code is able to access the input files either locally, or remotely via the OPeNDAP protocol. 
+
+## Method
+
+Generating function: 
+
+```
+usage: velocity_aggregated_timeseries.py [-h] -site SITE_CODE -files FILENAMES
+                                         [-indir INPUT_DIR]
+                                         [-outdir OUTPUT_DIR]
+                                         [-download_url DOWNLOAD_URL]
+                                         [-opendap_url OPENDAP_URL]
+
+Concatenate X,Y,Z velocity variables from ALL instruments from ALL deployments
+from ONE site
+
+optional arguments:
+  -h, --help                    show this help message and exit
+  -site SITE_CODE               site code, like NRMMAI
+  -files FILENAMES              name of the file that contains the source URLs
+  -indir INPUT_DIR              base path of input files
+  -outdir OUTPUT_DIR            path where the result file will be written. Default ./
+  -download_url DOWNLOAD_URL    path to the download_url_prefix
+  -opendap_url OPENDAP_URL      path to the opendap_url_prefix
+
+
+```
+
+
+
+### Input file validation
+
+Before proceeding to the aggregation, each input file will be checked to ensure it meets the requirements (as specified above under Inputs). Any input files that fail to meet the requirements will be excluded from the aggregation, and their URL listed in a global attribute `rejected_files`.
+
+### Dimensions
+
+The dimensions of the resulting file  are determined as follows:
+
+- `OBSERVATION`:    the total number of observation records, excluding out-of-the-water data, in all input files;
+- `INSTRUMENT`:     the number of instruments (i.e. number of files);
+- `strlen`:         a fixed dimension of length 256 for character array variables.
+
+### Variables
+
+Only values flagged as “good” or “probably good” are included. The velocity variables are produced by flattening, and concatenating the arrays in each of the input files. The time of the measurement is shifted according to the seconds_to_middle_of_measurement attribute. The values are then averaged into one-hour time bins (independently within each depth cell for ADCPs). The resulting variables have dimension `OBSERVATION`. 
```

Comment:
> The time of the measurement is shifted according to the seconds_to_middle_of_measurement attribute

This is currently _not_ done in the code. Need to make sure they're consistent when we're done.